aptos-move/move-examples/fungible_asset/sources/fungible_source.move

```diff
@@ -0,0 +1,268 @@
+module fungible_asset::fungible_source {
+
+    use aptos_framework::object::{object_address, Object, ExtendRef, address_from_extend_ref};
+    use std::option::Option;
+    use std::option;
+    use std::error;
+    use fungible_asset::fungible_asset::{mint, set_frozen_flag, withdraw_internal, burn, deposit};
+    use aptos_framework::object;
+    use std::signer;
+    #[test_only]
+    use fungible_asset::fungible_asset::{create_test_token, is_frozen};
+    #[test_only]
+    use aptos_framework::object::generate_extend_ref;
+    #[test_only]
+    use std::signer::address_of;
+
+    /// The fungible asset supply exists or does not exist for this asset object.
+    const EFUNGIBLE_SOURCE: u64 = 1;
+    /// Amount cannot be zero.
+    const EZERO_AMOUNT: u64 = 2;
+    /// The current_supply of token as fungible assets is not zero.
+    const ECURRENT_SUPPLY_NON_ZERO: u64 = 3;
+    /// Mint capability exists or does not exist.
+    const EMINT_CAP: u64 = 4;
+    /// Freeze capability exists does not exist.
+    const EFREEZE_CAP: u64 = 5;
+    /// Burn capability exists or does not exist.
+    const EBURN_CAP: u64 = 6;
+    /// Current supply overflow
+    const ECURRENT_SUPPLY_OVERFLOW: u64 = 7;
+    /// Current supply underflow
+    const ECURRENT_SUPPLY_UNDERFLOW: u64 = 8;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct FungibleSource has key {
+        current_supply: u64,
+        maximum_supply: Option<u64>
+    }
+
+    /// Capabilities
+    struct MintCap has store {
+        asset_addr: address
+    }
+
+    struct FreezeCap has store {
+        asset_addr: address
+    }
+
+    struct BurnCap has store {
+        asset_addr: address
+    }
```

Comment:
we should decide if we want to call these Refs or Caps. The new nomenclature is Refs.

I'd also recommend just adding drop to them, I'm not entirely a big fan with destroy, it adds somewhat arbitrary protections for accidentally deleting something....

Reply:
Cap for mint/burn, Ref for objects. Do you wanna mix the concept?

I am on the other side that I don't like `drop` to Caps exactly for the reason you mentioned. If ppl by accident forget a cap and don't mean it, it's gone. I prefer to let them deliberately destroy it at their own discretion.